File: catalog-server/src/catalog/image.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
pub enum ImageSize {
    DEFAULT,
    SMALL,
    LARGE,
    XLARGE,
}

#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
pub struct Image {
    pub id: String,
    pub size: ImageSize,
    pub width: u16,
    pub height: u16,
}

File: android/domain/src/main/java/com/lelloman/pezzottify/android/domain/remoteapi/response/Image.kt
package com.lelloman.pezzottify.android.domain.remoteapi.response

import kotlinx.serialization.Serializable

@Serializable
enum class ImageSize {
    Small,
    Default,
    Large,
    XLarge,
}

@Serializable
data class Image(
    val id: String,
    val uri: String,
    val size: ImageSize,
    val width: Int,
    val height: Int,
)